src/aro/Compilation.zig

```diff
@@ -614,15 +604,18 @@ fn generateSystemDefines(comp: *Compilation, w: anytype) !void {
     }
 }
 
-/// Generate builtin macros trying to use mtime as timestamp
-pub fn generateBuiltinMacrosFromPath(comp: *Compilation, system_defines_mode: SystemDefinesMode, path: []const u8) !Source {
-    const stat = comp.cwd.statFile(path) catch return try comp.generateBuiltinMacros(system_defines_mode, null);
-    const timestamp: i64 = @intCast(@divTrunc(stat.mtime, std.time.ns_per_s));
-    return try comp.generateBuiltinMacros(system_defines_mode, @intCast(std.math.clamp(timestamp, 0, max_timestamp)));
+pub fn loadSourceEpochFromEnvironment(comp: *Compilation) !void {
```

Comment:
Could you move this and the mtimes to the `Preprocessor`? Testing with Clang `SOURCE_DATE_EPOCH=abc gcc a.o` doesn't trigger the error so the behavior would match and it would avoid adding `FatalError` to `Compilation.initDefault()`.

Then again, it doesn't matter too much so use your own judgement.